diff --git a/projects/pr6-skeleton/tests/arrays/1d/arr-access-lvalue-multi.c b/projects/pr6-skeleton/tests/arrays/1d/arr-access-lvalue-multi.c
index cfe01ae..a34d152 100644
--- a/projects/pr6-skeleton/tests/arrays/1d/arr-access-lvalue-multi.c
+++ b/projects/pr6-skeleton/tests/arrays/1d/arr-access-lvalue-multi.c
@@ -4,5 +4,5 @@ void main()
     arr[3][3] = 42;
     int * p;
     p = &(arr[3][3]);
-    print *p;
+    write *p;
 }
diff --git a/projects/pr6-skeleton/tests/arrays/1d/arr-access-lvalue.c b/projects/pr6-skeleton/tests/arrays/1d/arr-access-lvalue.c
index d9eee11..5af8beb 100644
--- a/projects/pr6-skeleton/tests/arrays/1d/arr-access-lvalue.c
+++ b/projects/pr6-skeleton/tests/arrays/1d/arr-access-lvalue.c
@@ -4,5 +4,5 @@ void main()
     arr[3] = 42;
     int * p;
     p = &(arr[3]);
-    print *p;
+    write *p;
 }
diff --git a/projects/pr6-skeleton/tests/arrays/multi-d/2d.c b/projects/pr6-skeleton/tests/arrays/multi-d/2d.c
index 8634975..345aa31 100644
--- a/projects/pr6-skeleton/tests/arrays/multi-d/2d.c
+++ b/projects/pr6-skeleton/tests/arrays/multi-d/2d.c
@@ -10,7 +10,9 @@ void main()
 	second_index = 0;
 	while (second_index < 2) {
 	    arr[first_index][second_index] = first_index + second_index;
+	    second_index = second_index + 1;
 	}
+	first_index = first_index + 1;
     }
 
     first_index = 0;
@@ -18,6 +20,8 @@ void main()
 	second_index = 0;
 	while (second_index < 2) {
 	    write arr[first_index][second_index];
+	    second_index = second_index + 1;
 	}
+	first_index = first_index + 1;
     }
 }
diff --git a/projects/pr6-skeleton/tests/arrays/multi-d/5d.c b/projects/pr6-skeleton/tests/arrays/multi-d/5d.c
index 4eb6cdd..5ea8077 100644
--- a/projects/pr6-skeleton/tests/arrays/multi-d/5d.c
+++ b/projects/pr6-skeleton/tests/arrays/multi-d/5d.c
@@ -16,9 +16,13 @@ void main()
 		fourth = 0;
 		while (fourth < 2) {
 		    arr[first][second][third][fourth] = first + second + third + fourth;
+		    fourth = fourth + 1;
 		}
+	        third = third + 1;
 	    }
+	    second = second + 1;
 	}
+	first = first + 1;
     }
 
     first = 0;
@@ -30,8 +34,12 @@ void main()
 		fourth = 0;
 		while (fourth < 2) {
 		    write arr[first][second][third][fourth];
+		    fourth = fourth + 1;
 		}
+	        third = third + 1;
 	    }
+	    second = second + 1;
 	}
+ 	first = first + 1;
     }
 }
diff --git a/projects/pr6-skeleton/tests/basic-enhancements/logical-ops/or-short-circuit.c b/projects/pr6-skeleton/tests/basic-enhancements/logical-ops/or-short-circuit.c
index a53a006..43b298b 100644
--- a/projects/pr6-skeleton/tests/basic-enhancements/logical-ops/or-short-circuit.c
+++ b/projects/pr6-skeleton/tests/basic-enhancements/logical-ops/or-short-circuit.c
@@ -27,9 +27,9 @@ void main()
     }
 
     if (true || illegal(true)) {
-        write -1;
+        write 42;
     }
     else {
-        write 42;
+        write -1;
     }
 }
diff --git a/projects/pr6-skeleton/tests/basic-functions/plain-call-args.c b/projects/pr6-skeleton/tests/basic-functions/plain-call-args.c
index 18c015f..5482e99 100644
--- a/projects/pr6-skeleton/tests/basic-functions/plain-call-args.c
+++ b/projects/pr6-skeleton/tests/basic-functions/plain-call-args.c
@@ -1,5 +1,6 @@
 int foo(int x, int y)
 {
+    return 0;
 }
 
 void main()
diff --git a/projects/pr6-skeleton/tests/basic-functions/return-returns.c b/projects/pr6-skeleton/tests/basic-functions/return-returns.c
index e05d09c..4886395 100644
--- a/projects/pr6-skeleton/tests/basic-functions/return-returns.c
+++ b/projects/pr6-skeleton/tests/basic-functions/return-returns.c
@@ -10,7 +10,7 @@ int bar()
     write -1;
 }
 
-int main()
+void main()
 {
     foo();
     bar();
diff --git a/projects/pr6-skeleton/tests/floating-point/local.c b/projects/pr6-skeleton/tests/floating-point/local.c
index b16d029..92720ba 100644
--- a/projects/pr6-skeleton/tests/floating-point/local.c
+++ b/projects/pr6-skeleton/tests/floating-point/local.c
@@ -1,6 +1,6 @@
 void main()
 {
     double d;
-    d = 1.0;
+    d = 42.42;
     write d;
 }
diff --git a/projects/pr6-skeleton/tests/floating-point/relational.c b/projects/pr6-skeleton/tests/floating-point/relational.c
index cb52c10..79c52c5 100644
--- a/projects/pr6-skeleton/tests/floating-point/relational.c
+++ b/projects/pr6-skeleton/tests/floating-point/relational.c
@@ -15,7 +15,7 @@ void main()
         write 42;
     }
 
-    if (2.0<3.0) {
+    if (3.0<2.0) {
         write -1;
     }
     else {
@@ -38,7 +38,7 @@ void main()
         write -1;
     }
 
-    if (2.0<=3.0) {
+    if (3.0<=2.0) {
         write -1;
     }
     else {
diff --git a/projects/pr6-skeleton/tests/fundamentals/conditions.c b/projects/pr6-skeleton/tests/fundamentals/conditions.c
index 6abe956..a7f5bc7 100644
--- a/projects/pr6-skeleton/tests/fundamentals/conditions.c
+++ b/projects/pr6-skeleton/tests/fundamentals/conditions.c
@@ -15,7 +15,7 @@ void main()
         write 42;
     }
 
-    if (2<3) {
+    if (3<2) {
         write -1;
     }
     else {
@@ -38,7 +38,7 @@ void main()
         write -1;
     }
 
-    if (2<=3) {
+    if (3<=2) {
         write -1;
     }
     else {
diff --git a/projects/pr6-skeleton/tests/fundamentals/loop-with-if.c b/projects/pr6-skeleton/tests/fundamentals/loop-with-if.c
index 11f3551..9eda14e 100644
--- a/projects/pr6-skeleton/tests/fundamentals/loop-with-if.c
+++ b/projects/pr6-skeleton/tests/fundamentals/loop-with-if.c
@@ -2,9 +2,11 @@ void main()
 {
     int counter;
     int sum_of_evens;
+    int factorial;
 
     factorial = 1;
     counter = 0;
+    sum_of_evens = 0;
 
     while (counter < 6) {
         counter = counter + 1;
@@ -26,7 +28,7 @@ void main()
         }
     }
 
-    write (sum_of_evens + sum_of_evens);
+    write (sum_of_evens + 30);
 }
 
 
diff --git a/projects/pr6-skeleton/tests/fundamentals/loop.c b/projects/pr6-skeleton/tests/fundamentals/loop.c
index 547cc1c..1a6ed95 100644
--- a/projects/pr6-skeleton/tests/fundamentals/loop.c
+++ b/projects/pr6-skeleton/tests/fundamentals/loop.c
@@ -6,13 +6,16 @@ void main()
     factorial = 1;
     counter = 0;
 
-    while (counter < 4) {
+    while (counter < 3) {
         counter = counter + 1;
 
         int inner_counter;
         inner_counter = 0;
+	int old_factorial;
+	old_factorial = factorial;
         while (inner_counter < counter) {
-            factorial = factorial + counter;
+            factorial = factorial + old_factorial;
+            inner_counter = inner_counter + 1;
         }
     }
 
diff --git a/projects/pr6-skeleton/tests/fundamentals/pop-stack-after-stmt.c b/projects/pr6-skeleton/tests/fundamentals/pop-stack-after-stmt.c
index 72433b9..dca1b2b 100644
--- a/projects/pr6-skeleton/tests/fundamentals/pop-stack-after-stmt.c
+++ b/projects/pr6-skeleton/tests/fundamentals/pop-stack-after-stmt.c
@@ -4,7 +4,7 @@ void main()
 
     counter = 0;
 
-    while (counter < 1000000000) {
+    while (counter < 100000) {
         1;
         counter = counter + 1;
     }
diff --git a/projects/pr6-skeleton/tests/strings/output-literal.c b/projects/pr6-skeleton/tests/strings/output-literal.c
index 7324b9f..619e970 100644
--- a/projects/pr6-skeleton/tests/strings/output-literal.c
+++ b/projects/pr6-skeleton/tests/strings/output-literal.c
@@ -1,5 +1,5 @@
 void main()
 {
     write "This is a string\n";
-    write "This is another string\n";
+    write "This is another string";
 }
diff --git a/projects/pr6-skeleton/tests/strings/output-variable.c b/projects/pr6-skeleton/tests/strings/output-variable.c
index 012db0f..5a41cfa 100644
--- a/projects/pr6-skeleton/tests/strings/output-variable.c
+++ b/projects/pr6-skeleton/tests/strings/output-variable.c
@@ -4,6 +4,6 @@ void main()
     s = "I am a string literal!\n";
     write s;
 
-    s = "I am another string literal!\n";
+    s = "I am another string literal!";
     write s;
 }
diff --git a/projects/pr6-skeleton/tests/strings/subscript-strings.c b/projects/pr6-skeleton/tests/strings/subscript-strings.c
index 09c5fe1..0bb645d 100644
--- a/projects/pr6-skeleton/tests/strings/subscript-strings.c
+++ b/projects/pr6-skeleton/tests/strings/subscript-strings.c
@@ -3,10 +3,10 @@ void main()
     string s;
     s = "Did you know that 2 * 3 is 6?";
 
-    int x = s[21];
+    int x = s[20];
     write x;
 
-    write "Or that if p = &x, *p == x?"[2 * 10];
+    write "Or that if p = &x, *p == x?"[9 + 10];
 
     write "*"[0];
     write "*abc"[0];
